### PR TITLE
Cleanup bodies for processed messages

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

File: src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs [hunk 1]
```diff
@@ -138,11 +141,16 @@ ILearningTransportTransaction GetTransaction()
         {
             try
             {
-                await ProcessFile(transaction, nativeMessageId)
+                var wasCommitted =  await ProcessFile(transaction, nativeMessageId)
```
Comment (bording) [inline src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs]: nit: There's an extra space here before the `await`

File: src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs [hunk 2]
```diff
@@ -156,13 +164,13 @@ await receiveCircuitBreaker.Failure(ex)
...
 
-        async Task ProcessFile(ILearningTransportTransaction transaction, string messageId)
+        async Task<bool> ProcessFile(ILearningTransportTransaction transaction, string messageId)
         {
             try
             {
                 var message = await AsyncFile.ReadText(transaction.FileToProcess)
                     .ConfigureAwait(false);
-                var bodyPath = Path.Combine(path,".bodies", $"{messageId}.body.txt");
+                var bodyPath = Path.Combine(path, ".bodies", $"{messageId}.body.txt");
```
Comment (bording) [inline src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs]: This should be updated to use `bodyDir` and the new consts as well.

**Comment by SimonCropp** [comment]
@bording feedback addressed

**Comment by bording** [comment]
@SimonCropp Can you make it also use the `BodyFileSuffix` const?

**Comment by SimonCropp** [comment]
@bording done

File: src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs [hunk 3]
```diff
@@ -24,7 +24,10 @@ public Task Init(Func<MessageContext, Task> onMessage, Func<ErrorContext, Task<E
             transactionMode = settings.RequiredTransactionMode;
 
             path = Path.Combine(basePath, settings.InputQueue);
-            
+            Directory.CreateDirectory(Path.Combine(path, ".committed"));
```
Comment (bording) [inline src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs]: One last question, why has this line to create the committed dir been added here? I'm not seeing anywhere else in this PR that seems to need it?

Comment (andreasohlund) [inline src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs]: Pretty sure its not needed, will verify

**Comment by andreasohlund** [comment]
@bording good catch with the directory creation, was a misstake. Will merge once green since your comments are now addressed

**Comment by andreasohlund** [comment]
@timbussmann @dbelcham  this one is good to merge

**Pull request merged**